frontend/src/hooks/useFilters.js

```diff
@@ -0,0 +1,72 @@
+import { useCallback, useContext } from 'react';
```

Comment:
So should we be using this hook on any page that has filters?

Reply:
that's the idea, not going to plug these in everywhere just yet. less code to test.